src/gui/folderman.cpp

```diff
@@ -825,30 +828,52 @@ void FolderMan::slotStartScheduledFolderSync()
 
 void FolderMan::slotEtagPollTimerTimeout()
 {
-    ConfigFile cfg;
-    auto polltime = cfg.remotePollInterval();
+    const auto folderMapValues = _folderMap.values();
+    QList<Folder *> foldersToRun;
 
-    for (Folder *f : qAsConst(_folderMap)) {
-        if (!f) {
-            continue;
-        }
-        if (f->isSyncRunning()) {
-            continue;
-        }
-        if (_scheduledFolders.contains(f)) {
-            continue;
-        }
-        if (_disabledFolders.contains(f)) {
-            continue;
-        }
-        if (f->etagJob() || f->isBusy() || !f->canSync()) {
-            continue;
-        }
-        if (f->msecSinceLastSync() < polltime) {
-            continue;
-        }
-        QMetaObject::invokeMethod(f, "slotRunEtagJob", Qt::QueuedConnection);
+    // Some folders need not to be checked because they use the push notifications
+    std::copy_if(folderMapValues.begin(), folderMapValues.end(), foldersToRun.begin(), [this](Folder *folder) {
+        const auto account = folder->accountState()->account();
+        const auto capabilities = account->capabilities();
+        const auto pushNotifications = account->pushNotifications();
+
+        return (!pushNotifications || !pushNotifications->isReady());
```

Comment:
Occurred to me that now that we have more than one type of notification, we might want to check if the Files push notification type is supported? Or do we assume this one will always be there as long as you got notifications?

Reply:
Yes, this would be better.